webrecorder/browsertrix
# Crawls list backend pagination

Resolves #629, follows https://github.com/webrecorder/browsertrix-cloud/pull/736



### Changes

- Adds pagination to crawls list with >50 crawls (tested using fake data locally)
- Refactors state filter (i.e. status) and sort to use backend queries
- Enables backend searching for filters

### Manual testing

1. Log in to an account with >50 crawls and go to Crawls view. Verify pagination is visible at bottom of crawl list
2. Click through pagination controls. Verify controls work as expected
3. Update "View:" filter. Verify pagination is reset to page 1 and crawls update as expected
4. Type into "Filter by" search bar. Verify results show after typing in more than 1 character
5. Delete entry. Verify filter clears
6. Click a search filter result. Verify crawls are filtered as expected
7. Regression test: "Sort by" and "Show only mine" controls

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Crawls - Pagination controls | <img width="326" alt="Screen Shot 2023-04-03 at 1 15 33 PM" src="https://user-images.githubusercontent.com/4672952/229621304-dd751d94-0f58-4f88-8a4c-b42903098279.png"><img width="459" alt="Screen Shot 2023-04-03 at 1 22 36 PM" src="https://user-images.githubusercontent.com/4672952/229621312-6abfdc76-67d1-4b38-ab38-ad83d0fc5137.png"><img width="472" alt="Screen Shot 2023-04-03 at 1 22 42 PM" src="https://user-images.githubusercontent.com/4672952/229621314-6087aa18-3edc-4689-8fe9-343ecb07182b.png"> |
| Crawls - Search input - With results | <img width="545" alt="Screen Shot 2023-04-03 at 1 27 58 PM" src="https://user-images.githubusercontent.com/4672952/229621362-3e948795-eb43-467f-90cd-11933aa41ef0.png"> |
| Crawls - Search input - Filter selected | <img width="540" alt="Screen Shot 2023-04-03 at 1 28 04 PM" src="https://user-images.githubusercontent.com/4672952/229621410-dee2216e-1a02-44e7-af39-57915b3da508.png"><img width="535" alt="Screen Shot 2023-04-03 at 1 28 23 PM" src="https://user-images.githubusercontent.com/4672952/229621417-c642ebc9-b1e5-49b3-bb4d-6db5104b4cbd.png"> |
